src/utils.ts

```diff
@@ -97,7 +105,93 @@ export const getTracerInfo = (): { name: string; version: string } => {
   return { name, version };
 };
 
+const xRayFieldKeyValuePattern = /([^=;]+)=([^;]*)/g;
+
+export const splitXrayTraceIdToFields = (
+  awsXAmznTraceId: string,
+  maxFields: number = 100
+): { [key: string]: string } => {
+  const result: { [key: string]: string | null } = {};
+  let match: RegExpExecArray | null;
+  let iterations = 0;
+
+  // Resetting the regex lastIndex to ensure it works correctly for different inputs
+  xRayFieldKeyValuePattern.lastIndex = 0;
```

Comment:
Might be irrelevant if you switch to the abbreviated method above, but anyway - why is this required?

Reply:
removed, switched to the new implementation